### PR TITLE
[MM-38427] - Start Trial option showing in Daily

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
This PR takes care of an edge case where if the instance has never had a trial license used and currently has a non trial license loaded, the CTA shows up. This PR ensures that the CTA doesn't show in such a scenario.

#### Ticket Link
[MM-38427](https://mattermost.atlassian.net/browse/MM-38427)

#### Release Note
```release-note
NONE

```

### CODE REVIEW

**Comment by AGMETEOR** [comment]
@stevemudie QA steps are on the ticket

**Review by stevemudie — APPROVED** [review]
Nice work @AGMETEOR !

Everything worked as expected.
- The CTA was present without any license in place
- Once licensed the CTA was gone
- Removing the license brought the CTA back
- Starting a trial made it go away again
- Removing the trial license didn't bring the CTA back as expected and described in the Jira ticket QA steps

File: components/widgets/menu/menu_items/menu_start_trial.tsx [hunk 1]
```diff
@@ -33,8 +34,12 @@ const MenuStartTrial = (props: Props): JSX.Element | null => {
...
 
     const prevTrialLicense = useSelector((state: GlobalState) => state.entities.admin.prevTrialLicense);
-    const isLicensed = prevTrialLicense?.IsLicensed;
-    const show = isLicensed !== undefined && isLicensed !== 'true';
+    const currentLicense = useSelector(getLicense);
+    const isPrevLicensed = prevTrialLicense?.IsLicensed;
+    const isCurrentLicensed = currentLicense?.IsLicensed;
+
+    // Show this CTA if the instance is currently not licensed and has never had a trial license loaded before
+    const show = (isCurrentLicensed !== undefined && isCurrentLicensed !== 'true') && (isPrevLicensed !== undefined && isPrevLicensed !== 'true');
```
Comment (pablovelezvidal) [inline components/widgets/menu/menu_items/menu_start_trial.tsx:42]: Cant this be espresed as  x==='false'?

Comment (marianunez) [reply components/widgets/menu/menu_items/menu_start_trial.tsx:42]: +1, let's simplify this expression

**Review by marianunez — APPROVED** [review]
LGTM aside Pablos feedback

File: components/widgets/menu/menu_items/menu_start_trial.tsx (hunk 1 above)
Comment (AGMETEOR) [reply components/widgets/menu/menu_items/menu_start_trial.tsx:42]: Hi @pablovelezvidal it's possible that in the redux state we don't have `isLicensed` available while the license is being fetched. Would `x === 'false'` still work well?

**Review by pablovelezvidal — APPROVED** [review]
LGTM! thanks @AGMETEOR

**Pull request merged**